source/extensions/filters/http/rbac/rbac_filter.cc

```diff
@@ -44,7 +44,17 @@ Http::FilterHeadersStatus RoleBasedAccessControlFilter::decodeHeaders(Http::Head
   const Filters::Common::RBAC::RoleBasedAccessControlEngine& engine =
       config_->engine(callbacks_->route());
 
-  if (engine.allowed(*callbacks_->connection(), headers)) {
+  if (config_->config().has_permissive_rules()) {
+    if (engine.allowed(*callbacks_->connection(), headers,
+                       Filters::Common::RBAC::EnforcementMode::PERMISSIVE)) {
+      config_->stats().permissive_allowed_.inc();
+    } else {
+      config_->stats().permissive_denied_.inc();
+    }
+  }
```

Comment:
so this will report allow/deny ignoring the existing enforced rules. Is that the intended behavior?

Reply:
since enforced/permissive rules are independent, it should be expected.

Reply:
I guess my confusion here is should the user see the permissive rules as additive to the enforced policies, or as a replacement?

Reply:
It should be replacement. For example, we already have policy A in enforced mode running in prod, now want to test policy B; then in config, enforced_rules contains {A.mode==enforced}, permissive_rules contains {A.mode==permissive, B.mode=permissive}. This will allow to test policy replacement/deletion scenarios.  

I'll update the documentation, thanks.